README.md

```diff
@@ -1,2 +1,233 @@
+    values of  `OAuth Client Key` and `OAuth Client Secret` shown here.
+    ![OAuth Client Credentials](https://user-images.githubusercontent.com/15249242/91567068-27155e00-e962-11ea-8eab-b3bdd790bfd4.png)
+
+### Configuring the sample
+1. Download the sample application setup from [here](link).
+2. Double click the `setup.exe`.
+3. Follow the on-screen guidance until you get to the app configuration window.
+   ![forms](https://user-images.githubusercontent.com/15249242/95334388-a84e0200-08cb-11eb-89e6-e44d8ac158e4.gif)
+4. Fill out the following fields.
+
+       Client ID - <Enter the copied value of `OAuth Client Key` when creating the Service Provider>
+       Client Secret - <Enter the copied value of `OAuth Client Secret` when creating the Service Provider>
+       Authorization Endpoint - https://localhost:9443/oauth2/authorize
+       Token Endpoint - https://localhost:9443/oauth2/token
+       Userinfo Endpoint - https://localhost:9443/oauth2/userinfo
+       Logout Endpoint - https://localhost:9443/oidc/logout
+       Redirect URI - http://localhost:8080/callback/
+       PostLogout Redirect URI - http://localhost:8080/postlogout/
+5. Continue the on-screen guidance and complete the installation.
+
+### Running the sample
+Double click on the `Pickup Manager` application available on your Desktop.<br/>
+![pickup manager](https://user-images.githubusercontent.com/15249242/95334396-aa17c580-08cb-11eb-83ee-3b88b8512f68.gif)
+
+## How it works
+This section explains detailed walkthrough on how key aspects handled in the Asgardio .NET OIDC SDK.
+Througout this section we will refer to the source folder of the sample as <APP_HOME>
+
+  * [Trigger authentication](#trigger-authentication)
+  * [Retrieve user attributes](#retrieve-user-attributes)
+  * [Trigger logout](#trigger-logout)
+
+The structure of the sample would be as follows:<br/>
+![structure](https://user-images.githubusercontent.com/15249242/95333060-e34f3600-08c9-11eb-91e3-8ea64446eb77.PNG)
+
+### Trigger authentication
+In the `<APP_HOME>/LoginPage.xaml` page, we have registered a `Click` event named `LoginButton_Click` for the login button to trigger a OIDC authentication:
+```xml
+<Button x:Name ="login" Content="LOGIN" HorizontalAlignment="Left" Margin="73,452,0,0" VerticalAlignment="Top" 
+                Width="240" Height="56" FontFamily="Segoe UI Black" FontSize="22" Click="LoginButton_Click" Background="LightSeaGreen" Foreground="Black"/>
+```
+
+The button click would trigger an authentication request, and redirect the user to the IdP authentication page.
+Upon successful authentication, the user would be redirected to the application homepage.
+
+### Retrieve user attributes
+In the `<APP_HOME>/LoginPage.xaml.cs` file, we have added the following code inside the login button click trigger method to get the user subject value and the user attributes referring the SDK API.
```

Comment:
Would be more readable if the "login button click" trigger method was differentiated from the rest of the text. Maybe we could use the method name itself like `LoginButton_Click`.

Reply:
Resolved in https://github.com/asgardio/asgardio-dotnet-oidc-sdk/pull/4/commits/3945c817476dfe36be1ba0eac68d777f7d6dd006